src/core.js

```diff
@@ -821,6 +821,7 @@ $.extend( $.validator, {
 					group = this.groups[ element.name ];
 					if ( group ) {
 						$.each( this.groups, function( name, testgroup ) {
+							name = name.replace( /'/g, "\\'" );
 							if ( testgroup === group ) {
```

Comment:
Maybe dven move it into the selector expression instead of re-assign the local var